packages/preferences/src/node/json-preference-server.ts

```diff
@@ -100,27 +101,39 @@ export class JsonPreferenceServer implements PreferenceServer {
             if (this.preferences) {
                 this.fireChanged(this.preferences, preferences);
             } else {
-                this.fireNew(preferences);
+                // Will not fire individual events until the server has finished reading the preferences once
+                if (this.isServerReady) {
+                    this.fireNew(preferences);
+                }
             }
         } else if (this.preferences) {
             this.fireRemoved(this.preferences);
         }
         this.preferences = preferences;
+        // Send an event containing all preferences once
+        if (!this.isServerReady) {
+            this.fireInitEvent();
+        }
+        this.isServerReady = true;
+    }
+
+    protected fireInitEvent(): void {
+        this.fireEvent({ preferenceName: "", newValue: this.preferences });
     }
 
     protected fireNew(preferences: any): void {
         // tslint:disable-next-line:forin
         for (const preferenceName in preferences) {
             const newValue = preferences[preferenceName];
-            this.fireEvent({ preferenceName, newValue });
+            this.fireEvent({ preferenceName: preferenceName, newValue: newValue });
```

Comment:
I meant that we don't send an individual events anymore, but only one `this.fireEvent({changes})` of this shape:
```ts
export interface PreferenceChangedEvent {
    changes: PreferenceChange[]
}

export interface PreferenceChange {
    readonly preferenceName: string;
    readonly newValue?: any;
    readonly oldValue?: any;
}
```